src/util/server_table.c

```diff
@@ -42,53 +47,62 @@ int server_table_add(charbuf server_id, uint64_t handle)
     return MEM_ALLOC_FAIL;
   }
 
-  tmp_entry.id = new_charbuf(server_id.len);
-  if (server_id.len != tmp_entry.id.len)
-  {
-    pelz_log(LOG_ERR, "Charbuf creation error.");
-    return ERR_BUF;
-  }
-
-  memcpy(tmp_entry.id.chars, server_id.chars, tmp_entry.id.len);
   data_size = retrieve_from_unseal_table(handle, &data);
   if (data_size == 0)
   {
     pelz_log(LOG_ERR, "Failure to retrive data from unseal table.");
     return RET_FAIL;
   }
 
-  tmp_entry.value.cert = new_charbuf(data_size);
-  if (data_size != tmp_entry.value.cert.len)
+  ret = unmarshal_ec_der_to_x509(&data, &data_size, &tmp_entry.value.cert);
+  if (ret)
   {
-    pelz_log(LOG_ERR, "Charbuf creation error.");
-    return ERR_BUF;
+    pelz_log(LOG_ERR, "Unmarshal DER to X509 Failure");
+    free(data);
+    return ERR_X509;
   }
+  free(data);
 
-  memcpy(tmp_entry.value.cert.chars, data, tmp_entry.value.cert.len);
+  X509_NAME *subj = X509_get_subject_name(tmp_entry.value.cert);
```

Comment:
Is it possible for this call to fail, and if so, can we detect it immediately?

Reply:
Check added